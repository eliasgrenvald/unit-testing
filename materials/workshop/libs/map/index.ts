export * from './base';
export * from './models';
export * from './services';
