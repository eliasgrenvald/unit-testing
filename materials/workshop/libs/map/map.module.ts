import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { Angulartics2Module } from 'angulartics2';
import { MapService } from './services';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import * as fromMapReducers from '@maersk-digital/root-features/map/store/reducers';
import * as fromMapEffects from '@maersk-digital/root-features/map/store/effects';

@NgModule({
	imports:
		[
			CommonModule,
			Angulartics2Module,
			StoreModule.forFeature('map', fromMapReducers.mapReducers),
			EffectsModule.forFeature([
				...fromMapEffects.MAP_EFFECTS
			])
		],
	providers:
		[
			MapService
		]
})
export class MapModule {}
