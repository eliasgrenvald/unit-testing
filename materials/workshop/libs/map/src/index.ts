export * from './lib/map.module';
