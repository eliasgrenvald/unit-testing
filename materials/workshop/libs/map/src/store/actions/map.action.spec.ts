import { Feature, Point } from 'geojson';
import { SelectMapFeature } from './map.action';

describe('SelectMapFeature action', () => {
	test('should set the payload', () => {
		const point = <Feature<Point>>{
			type: 'Feature',
			geometry:
				{
					type: 'Point',
					coordinates:
						[
							10.263290405273438,
							56.17155205794858
						]
				}
		};

		const action = new SelectMapFeature(point);

		expect(action.payload).toMatchSnapshot();
	});
});
