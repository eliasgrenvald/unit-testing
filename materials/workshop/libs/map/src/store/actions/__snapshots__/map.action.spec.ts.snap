// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMapFeature action should set the payload 1`] = `
Object {
  "geometry": Object {
    "coordinates": Array [
      10.263290405273438,
      56.17155205794858,
    ],
    "type": "Point",
  },
  "type": "Feature",
}
`;
